File: backend/timebank/src/main/java/kookmin/software/capstone2023/timebank/application/service/account/AccountReadService.java
package kookmin.software.capstone2023.timebank.application.service.account;

import kookmin.software.capstone2023.timebank.application.exception.NotFoundException;
import kookmin.software.capstone2023.timebank.domain.model.Account;
import kookmin.software.capstone2023.timebank.domain.repository.AccountJpaRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.Optional;

@Service
public class AccountReadService {
    private final AccountJpaRepository accountRepository;

    @Autowired
    public AccountReadService(AccountJpaRepository accountRepository) {
        this.accountRepository = accountRepository;
    }

    public Optional<Account> getAccountById(Long accountId) {
        Optional<Account> account = accountRepository.findById(accountId);
        if (account.isPresent()) {
            throw new NotFoundException("Account not found");
        }
        return account;
    }
}

File: src/main/java/kookmin/software/capstone2023/timebank/application/service/account/AccountReadService.kt
package kookmin.software.capstone2023.timebank.application.service.account

import kookmin.software.capstone2023.timebank.application.exception.NotFoundException
import kookmin.software.capstone2023.timebank.domain.model.Account
import kookmin.software.capstone2023.timebank.domain.repository.AccountJpaRepository
import org.springframework.data.repository.findByIdOrNull

class AccountReadService(
    private val accountRepository: AccountJpaRepository,
) {
    fun getAccountById(
        accountId: Long,
    ): Account {
        return accountRepository.findByIdOrNull(accountId)
            ?: throw NotFoundException(message = "Account not found")
    }
}
